seatunnel-plugin-discovery/src/main/java/org/apache/seatunnel/plugin/discovery/AbstractPluginDiscovery.java

```diff
@@ -75,12 +76,24 @@ public abstract class AbstractPluginDiscovery<T> implements PluginDiscovery<T> {
      * Add jar url to classloader. The different engine should have different logic to add url into
      * their own classloader
      */
-    private static final BiConsumer<ClassLoader, URL> DEFAULT_URL_TO_CLASSLOADER =
+    protected static final BiConsumer<ClassLoader, URL> DEFAULT_URL_TO_CLASSLOADER =
             (classLoader, url) -> {
-                if (classLoader instanceof URLClassLoader) {
-                    ReflectionUtils.invoke(classLoader, "addURL", url);
-                } else {
-                    throw new UnsupportedOperationException("can't support custom load jar");
+                try {
+                    if (classLoader.getClass().getName().endsWith("SafetyNetWrapperClassLoader")) {
```

Comment:
It only worked for flink, we should not put `SafetyNetWrapperClassLoader` check at here.